src/server/http_api.cc

```diff
@@ -0,0 +1,228 @@
+    absl::StrAppend(&str, v);
+  }
+
+  void operator()(double v) {
+    absl::StrAppend(&str, v);
+  }
+
+  void operator()(const CapturingReplyBuilder::SimpleString& ss) {
+    absl::StrAppend(&str, "\"", ss, "\"");
+  }
+
+  void operator()(const CapturingReplyBuilder::BulkString& bs) {
+    absl::StrAppend(&str, JsonEscape(bs));
+  }
+
+  void operator()(CapturingReplyBuilder::Null) {
+    absl::StrAppend(&str, "null");
+  }
+
+  void operator()(CapturingReplyBuilder::Error err) {
+    str = absl::StrCat(R"({"error": ")", err.first);
+  }
+
+  void operator()(facade::OpStatus status) {
+    absl::StrAppend(&str, "\"", facade::StatusToMsg(status), "\"");
+  }
+
+  void operator()(const CapturingReplyBuilder::StrArrPayload& sa) {
+    absl::StrAppend(&str, "not_implemented");
+  }
+
+  void operator()(unique_ptr<CapturingReplyBuilder::CollectionPayload> cp) {
+    if (!cp) {
+      absl::StrAppend(&str, "null");
+      return;
+    }
+    if (cp->len == 0 && cp->type == facade::RedisReplyBuilder::ARRAY) {
+      absl::StrAppend(&str, "[]");
+      return;
+    }
+
+    absl::StrAppend(&str, "[");
+    for (auto& pl : cp->arr) {
+      visit(*this, std::move(pl));
+    }
+  }
+
+  void operator()(facade::SinkReplyBuilder::MGetResponse resp) {
+    absl::StrAppend(&str, "not_implemented");
+  }
+
+  void operator()(const CapturingReplyBuilder::ScoredArray& sarr) {
+    absl::StrAppend(&str, "[");
+    for (const auto& [key, score] : sarr.arr) {
+      absl::StrAppend(&str, "{", JsonEscape(key), ":", score, "},");
+    }
+    if (sarr.arr.size() > 0) {
+      str.pop_back();
+    }
+    absl::StrAppend(&str, "]");
+  }
+
+  string str;
+};
+
+}  // namespace
+
+void HttpAPI(const http::QueryArgs& args, HttpRequest&& req, Service* service,
+             HttpContext* http_cntx) {
+  auto& body = req.body();
+
+  flexbuffers::Builder fbb;
+  flatbuffers::Parser parser;
+  flexbuffers::Reference doc;
+  bool success = parser.ParseFlexBuffer(body.c_str(), nullptr, &fbb);
+  if (success) {
+    fbb.Finish();
+    doc = flexbuffers::GetRoot(fbb.GetBuffer());
+    if (!IsVectorOfStrings(doc)) {
+      success = false;
+    }
+  }
+
+  if (!success) {
+    auto response = http::MakeStringResponse(h2::status::bad_request);
+    http::SetMime(http::kTextMime, &response);
+    response.body() = "Failed to parse json\r\n";
+    http_cntx->Invoke(std::move(response));
+    return;
+  }
+
+  vector<string> cmd_args;
+  flexbuffers::Vector vec = doc.AsVector();
+  for (size_t i = 0; i < vec.size(); ++i) {
+    cmd_args.push_back(vec[i].AsString().c_str());
+  }
+  vector<facade::MutableSlice> cmd_slices(cmd_args.size());
+  for (size_t i = 0; i < cmd_args.size(); ++i) {
+    cmd_slices[i] = absl::MakeSpan(cmd_args[i]);
+  }
+
+  facade::ConnectionContext* context = (facade::ConnectionContext*)http_cntx->user_data();
+  DCHECK(context);
+
+  facade::CapturingReplyBuilder reply_builder;
+  auto* prev = context->Inject(&reply_builder);
+  // TODO: to finish this.
```

Comment:
It'd be nice if you added _what's_ missing :)

Reply:
oh, it's a stale comment. I will remove it next time I touch this code.